### PR TITLE
Limit the grade of the answer to the question's maximum grade.

### PR DESCRIPTION
**Pull request by lowjoel**

### CODE REVIEW

File: app/models/course/assessment/answer.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@ class Course::Assessment::Answer < ActiveRecord::Base
   validate :validate_assessment_state, if: :attempting?
   validates :grade, presence: true, unless: :attempting?
   validates :grade, absence: true, if: :attempting?
+  validate :validate_consistent_grade, unless: :attempting?
```
Comment (allenwq) [inline app/models/course/assessment/answer.rb]: I think `validates_numericality_of` will work for this case ?

Comment (lowjoel) [inline app/models/course/assessment/answer.rb]: check the definition of the function

Comment (allenwq) [inline app/models/course/assessment/answer.rb]: `validates :grade, allow_nil: true, numericality: { less_than_or_equal_to: :maximum_grade }` Does this work ?

Comment (lowjoel) [inline app/models/course/assessment/answer.rb]: maximum grade is on the question not on the answer

Comment (allenwq) [inline app/models/course/assessment/answer.rb]: how about delegate it ?

Comment (lowjoel) [inline app/models/course/assessment/answer.rb]: an answer does not have a maximum grade, it only has a grade.

Comment (allenwq) [inline app/models/course/assessment/answer.rb]: OK, I'll follow your decision then.

Comment (lowjoel) [inline app/models/course/assessment/answer.rb]: thanks. this kind of graph structures aren't fantastic for SQL+ORM

**Pull request merged**